### PR TITLE
purge resource actions

### PR DESCRIPTION
**Pull request by arnaudsjs**
* Regularly purge old resource_action in database
* Also write resource-actions logs to file

### CODE REVIEW

File: src/inmanta/data.py [hunk 1]
```diff
@@ -573,6 +574,8 @@ class Environment(BaseDocument):
                                           doc="Agent interval for autostarted agents in seconds", agent_restart=True),
         SERVER_COMPILE: Setting(name=SERVER_COMPILE, default=True, typ="bool",
                                 validator=convert_boolean, doc="Allow the server to compile the configuration model."),
+        RESOURCE_ACTION_LOGS_RETENTION: Setting(name=RESOURCE_ACTION_LOGS_RETENTION, default=7, typ="int",
+                                                validator=convert_int, doc="Amount of days to retain resource-action logs"),
```
Comment (bartv) [inline src/inmanta/data.py]: This should be "Number of". Days is countable: https://dictionary.cambridge.org/grammar/british-grammar/amount-of-number-of-or-quantity-of

Comment (bartv) [reply src/inmanta/data.py]: I approved but this should be changed actually :)

File: src/inmanta/server/server.py [hunk 2]
```diff
@@ -101,6 +107,32 @@ def start(self):
...
+        yield self._close_resource_action_logger()
+
+    @gen.coroutine
+    def _create_resource_action_logger(self):
+        resource_action_log = os.path.join(opt.log_dir.get(), opt.server_resource_action_log.get())
+        self._file_handler = logging.handlers.WatchedFileHandler(filename=resource_action_log, mode='a+')
+        formatter = logging.Formatter(fmt="%(asctime)s %(levelname)-8s %(message)s")
+        self._file_handler.setFormatter(formatter)
+        self._file_handler.setLevel(logging.DEBUG)
+        self._resource_action_logger = logging.getLogger("RESOURCE_ACTION_LOG_FILE")
```
Comment (wouterdb) [inline src/inmanta/server/server.py]: looks a lot like a constant and should probably be a constant

the name of the logger is not that well chose, something like "resource_action_logger" for example would better describe it.

File: src/inmanta/server/server.py [hunk 3]
```diff
@@ -1157,6 +1191,10 @@ def resource_action_update(self, env, resource_ids, action_id, action, started,
 
         if len(messages) > 0:
             resource_action.add_logs(messages)
+            for msg in messages:
+                loglevel = const.LogLevel[msg["level"]].value
+                log_line = data.LogLine.log(loglevel, msg["msg"], *msg["args"])
+                self._write_to_resource_action_log(log_line)
```
Comment (wouterdb) [inline src/inmanta/server/server.py]: I'm quite curious about the performance impact of this. 
also about the latency wrt to flushing: does this flush to disk on every write?

Comment (arnaudsjs) [reply src/inmanta/server/server.py]: Yes. The data is flushed to disk on every write. This happens in the StreamHandler class, which is a superclass of WatchedFileHandler (https://github.com/python/cpython/blob/master/Lib/logging/__init__.py).

**Pull request merged**